### DISCUSSION TITLE
Modern PNP webpart Filter pages by language

### DISCUSSION THREAD

**Discussion by ValiPrk** [microsoft-search/pnp-modern-search#3847] [Q&A]
Hello,

I'm trying to filter some news pages by language when using the "SPTranslationLanguage" property, but translation is enabled and for the default language, the column is not filled-in with values, for the rest of the translated pages the values are set correctly, is there any way to overcome this?
Is there any other property that we can use?

Thank you very much for your help!

**Comment by kasperbolarsen** [comment]
In relation to the SharePoint Premium document translation service, I have asked the Microsoft product team the same question. While we wait for that change ( if it is approved ), I am afraid the only solution I can think off will require you make a workaround where a timer or event triggered job update a column on the new item :-(

**Comment by ValiPrk** [comment]
Do you have an article on how can I do that? From what I know that is an automatically generated column.
Thank you very much for your help!

**Comment by ValiPrk** [comment]
The "Item language" column is automatically generated and it is filled-in with the value of the name of the folder.
![Item language](https://github.com/microsoft-search/pnp-modern-search/assets/147488275/b63b8ec6-e259-47d1-ba55-ce844d7fc940)
